packages/orchestrator-ui-components/src/pages/processes/WfoProcessDetailPage.tsx

```diff
@@ -64,7 +70,22 @@ export const WfoProcessDetailPage = ({
         );
     }, [data, processDetailRefetchInterval]);
 
-    const process = data?.processes.page[0];
+    useEffect(() => {
+        const fetchedProcessDetails = data?.processes.page[0];
+
+        if (!process) {
+            setProcess(fetchedProcessDetails);
+            return;
+        }
+
+        const updateProcess =
+            process.lastStatus != fetchedProcessDetails?.lastStatus ||
+            process.lastStep !== fetchedProcessDetails?.lastStep;
+        if (updateProcess) {
+            setProcess(fetchedProcessDetails);
+        }
+    }, [data]);
+
```

Comment:
I think it might be better to disable the refetching from the useQueryWithGraphql hook instead of introducing more state.

Reply:
This doesn't disable refetching though right? It only determines if an update should take place

Reply:
every interval the data updates and is seen as a state change, updating everything. this doesn't disable fetching interval but changes it so the form components don't rerender each interval.